### ISSUE TITLE
Modify the "Create" method to work with "roles"

### ISSUE THREAD

**Issue by bsaiken** [FWAJL/FieldWorkAssistantMVC#809]
Once the admin role (Issue #421) is set up, we can have the "Create" method work only within the admin role.  Since we are early in the development, I want the admin to have complete control over adding a new PM.

**Comment by bsaiken** [comment]
@damirius @FWAJL 
Add a method that allows us to create new users.  There are several roles, so this describes how each user type gets created, but I think the method can be written to be dynamic, and then just pass the variable for what kind of user needs to be made.
Admin: There is no need to make an admin user.  As Damir has said, all we need to do it take one of the PM users and change the user type to admin.  There is only one admin, so this awkward way of creating the user is OK.  I will make another Issue to allow the admin be able to change their password.
PM: This user type is created by the admin.  I will make a new Issue for this.
Field Technician: The FT user type is created every time a new technician is added.  The technician will be sent an email with their temporary username and password (which, in both cases, will be their email).
Service Provider: This will be a backburner Issue.  I'm not sure how we will implement this just yet.
Client: This will also be a backburner Issue.  Not sure how this will be implemented yet.

**Comment by damirius** [comment]
@bsaiken @FWAJL 

So essentially for now we need ability to create new PMs by admin user(s) since FTs are created automatically? We have currently one shared table `users` which is just the login info and few others for specific type of users, for now those are `project_manager` table and `technician` table (we have `service`,`client` tables but I'm not counting them just yet). Since those tables have different fields, we will use different forms for each user type. We can maybe show default form for `user` table since that table is shared accross all user types. It will contain text fields for username, password and hint and dropdown menu for user type/role. Just want to say here that we actually have separate type and role, so user can be created with the type "pm_id" and linked to some `project_manager` record but with "Field Technician" role. I know that we have one type for each role, so type "pm_id" would be role "Project manager" so I can set up there just one dropdown instead of two separate ones.

**Comment by FWAJL** [comment]
> FTs are created automatically

@damirius : I think Brian wasn't clear enough: we need to code this still. So when you add a FT, (1) you add a row in `technician`table, just like normal and (2) then you add a row in the `user` table with the login = email, password = hashed email (same hashing as the pm password), user_type = "technician_id", user_value = technician_id value created before, user_role_id = 3.
(2) is the point to do regarding the FT.

> Just want to say here that we actually have separate type and role, so user can be created with the type "pm_id" and linked to some project_manager record but with "Field Technician" role. I know that we have one type for each role, so type "pm_id" would be role "Project manager" so I can set up there just one dropdown instead of two separate ones.

@bsaiken : Why would we let the user update his role? It is ok to have if you're logged in as an admin but not as PM or FT. To what role would a PM set a FT ? Maybe, we should leave this field non editable to avoid bugs for now.

**Comment by damirius** [comment]
@FWAJL 

I understand that we still need to create actual user for the technician, but that's automatic creation of user info for technicians when we add them over our existing form "technician/showForm".
This page for adding new users will be restricted to admins only, so PM can't change roles of FTs. They will be created on different pages.

**Comment by bsaiken** [comment]
@damirius @FWAJL 

Jeremie's explanation is correct.  Except, it does not belong here.  The explanation belongs to Issue #812, so I copied the text there.

No users will be able to modify their role.

What we really need is for Issue #811 to be completed.  In order to complete Issue #811, there needs to be a method to add a (PM) user.  I was just mentioning above that when you write the method, make it dynamic, so that when we get to Issue #812, the method is ready to use.

**Comment by FWAJL** [comment]
@bsaiken 

What is the purpose of this issue since #811 takes care of adding (or creating) PMs and #812 takes care of adding FTs?

Also, is #812 not really important to start the Mobile tool building? I think #812 should follow straight after #811

**Comment by damirius** [comment]
@bsaiken @FWAJL 

See #811

**Comment by damirius** [comment]
@bsaiken @FWAJL 

Since we split up some issues, can you tell me what's the purpose of this one? I think Jeremie already asked that few days ago.

**Comment by bsaiken** [comment]
Yes, I think that this Issue could be closed. I have left it open because there is a lot if valuable discussions.

**Comment by FWAJL** [comment]
@bsaiken 

This can be closed as I labelled it "Documentation". This is how we can find it once closed.
